x-pack/plugins/infra/server/plugin.ts

```diff
@@ -4,30 +4,149 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { Plugin, PluginInitializerContext } from 'src/core/server';
+import { CoreSetup, PluginInitializerContext } from 'src/core/server';
+import { Server } from 'hapi';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+import { initInfraServer } from './infra_server';
+import { InfraBackendLibs, InfraDomainLibs } from './lib/infra_types';
+import { FrameworkFieldsAdapter } from './lib/adapters/fields/framework_fields_adapter';
+import { KibanaFramework } from './lib/adapters/framework/kibana_framework_adapter';
+import { InfraKibanaLogEntriesAdapter } from './lib/adapters/log_entries/kibana_log_entries_adapter';
+import { KibanaMetricsAdapter } from './lib/adapters/metrics/kibana_metrics_adapter';
+import { InfraElasticsearchSourceStatusAdapter } from './lib/adapters/source_status';
+import { InfraFieldsDomain } from './lib/domains/fields_domain';
+import { InfraLogEntriesDomain } from './lib/domains/log_entries_domain';
+import { InfraMetricsDomain } from './lib/domains/metrics_domain';
+import { LogEntryCategoriesAnalysis, LogEntryRateAnalysis } from './lib/log_analysis';
+import { InfraSnapshot } from './lib/snapshot';
+import { InfraSourceStatus } from './lib/source_status';
+import { InfraSources } from './lib/sources';
+import { InfraServerPluginDeps } from './lib/adapters/framework';
+import { METRICS_FEATURE, LOGS_FEATURE } from './features';
+import { UsageCollector } from './usage/usage_collector';
+import { InfraStaticSourceConfiguration } from './lib/sources/types';
 
-export class InfraPlugin implements Plugin<InfraSetup> {
-  private readonly initContext: PluginInitializerContext;
+export const config = {
+  schema: schema.object({
+    enabled: schema.maybe(schema.boolean()),
+    query: schema.object({
+      partitionSize: schema.maybe(schema.number()),
+      partitionFactor: schema.maybe(schema.number()),
+    }),
+  }),
```

Comment:
Isn't this incomplete? Shouldn't this be imported or copied from `kibana.index.ts`?

Attempting to start with anything like `xpack.infra.sources.default.logAlias: 'filebeat-empty'` in the config now fails.

Reply:
I don't know as much about the configuration handling as @jasonrhodes handled that, I'll investigate.

Reply:
Ah, yeah we probably need to merge these two config situations so we only have one place ... I think we can move things over from the legacy file to this new schema set up now that we are out of the legacy directory. Good catch.